webapp/src/components/cardDialog.tsx

```diff
@@ -114,40 +120,115 @@ const CardDialog = (props: Props): JSX.Element => {
             onClickDelete={handleDeleteButtonOnClick}
         >
             {!isTemplate &&
-                <BoardPermissionGate permissions={[Permission.ManageBoardProperties]}>
-                    <Menu.Text
-                        id='makeTemplate'
-                        icon={
-                            <CompassIcon
-                                icon='plus'
-                            />}
-                        name='New template from card'
-                        onClick={makeTemplateClicked}
-                    />
-                </BoardPermissionGate>
+            <BoardPermissionGate permissions={[Permission.ManageBoardProperties]}>
+                <Menu.Text
+                    id='makeTemplate'
+                    icon={
+                        <CompassIcon
+                            icon='plus'
+                        />}
+                    name='New template from card'
+                    onClick={makeTemplateClicked}
+                />
+            </BoardPermissionGate>
             }
         </CardActionsMenu>
     )
 
-    const followActionButton = (following: boolean): React.ReactNode => {
-        const followBtn = (
+    const selectAttachment = (boardId: string) => {
+        return new Promise<AttachmentBlock>(
+            (resolve) => {
+                Utils.selectLocalFile(async (attachment) => {
+                    const attachmentId = await octoClient.uploadFile(boardId, attachment)
+                    if (attachmentId) {
+                        const block = createAttachmentBlock()
+                        block.fields.attachmentId = attachmentId || ''
+                        resolve(block)
+                    } else {
+                        sendFlashMessage({content: intl.formatMessage({id: 'createFileBlock.failed', defaultMessage: 'Unable to upload the file. File size limit reached.'}), severity: 'normal'})
+                    }
+                },
+                '')
+            },
+        )
+    }
+
+    const addElement = async () => {
+        if (!card) {
+            return
+        }
+        const block = await selectAttachment(board.id)
+        block.parentId = card.id
+        block.boardId = card.boardId
+        const typeName = block.type
+        const description = intl.formatMessage({id: 'ContentBlock.addElement', defaultMessage: 'add {type}'}, {type: typeName})
+        await mutator.performAsUndoGroup(async () => {
+            const afterRedo = async (newBlock: Block) => {
+                const contentOrder = card.fields.contentOrder.slice()
+                contentOrder.splice(card.fields.contentOrder.length, 0, newBlock.id)
+                await octoClient.patchBlock(card.boardId, card.id, {updatedFields: {contentOrder}})
+            }
+
+            const beforeUndo = async () => {
+                const contentOrder = card.fields.contentOrder.slice()
```

Comment:
If this is necessary, you may want to check this, I believe this will be the new contentOrder as you are setting it above.